addons/web/static/tests/views/fields/radio_field.test.js

```diff
@@ -261,3 +268,86 @@ test("radio field is empty", async () => {
     expect(".o_radio_input:disabled").toHaveCount(3);
     expect(".o_radio_input:checked").toHaveCount(0);
 });
+
+test("radio selection field with filter, empty list", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <field name="color" widget="radio_selection_with_filter"
+                       context="{'allowed_selection': []}" />
+            </form>
+        `
+    })
+
+    expect(".o_radio_input").toHaveCount(0);
+});
+
+test("radio selection field with filter, single choice", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <field name="color" widget="radio_selection_with_filter"
+                       context="{'allowed_selection': ['black']}" />
+            </form>
+        `
+    })
+
+    expect(".o_radio_input").toHaveCount(1);
+    expect(".o_field_widget").toHaveText("Black");
+});
+
+test("radio selection field with filter, all choices", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <field name="color" widget="radio_selection_with_filter"
+                       context="{'allowed_selection': ['black', 'red']}" />
+            </form>
+        `
+    })
+
+    expect(".o_radio_input").toHaveCount(2);
+    expect(".o_field_widget").toHaveText("Red\nBlack");
+});
+
+test("radio selection field with filter, sync with other field", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <group>
+                    <field name="mood" widget="radio" />
+                    <field name="color" widget="radio_selection_with_filter"
+                           context="{'allowed_selection':
+                                ['red']
+                                + (['black'] if mood == 'sad' else [])
```

Comment:
Maybe a more realistic use case would be to have just `'allowed_selection': mood` here and adjust the value of "mood".

We would have to make "mood" a json field though, not sure how that's feasible.

Reply:
I agree that this test is a bit out of the blue. It does actually test a usecase I have in another PR: https://github.com/odoo/odoo/pull/158535 Like you I wanted to add another dynamic test but like you I wasn't too sure of how to do it, maybe the js people will know :)

Reply:
@awa-odoo I decided to give it a shot and to write a still-dynamic-but-less-crazy test. I decided to use two fields, a bool on and a selection one with the selection having a context depending on the boolean field. That test showed an important limitation.

When using two selection fields, each with its own radio buttons. Whenever you update one of the the radios, the other radio widget is updated (at least its `extractProps` method is called again). This is great because it uses the newer context.

When using a selection field and a boolean one. Updating the checkbox (of the boolean field) doesn't trigger `extractProps` on the selection widget. i.e. the widget isn't updated with the newer context, i.e. the selection doesn't change.

The thing is, in saas-17.2 all the places where this widget is used, the filtering context comes from computed fields server-side. I guess that those contextes are enough static to not cause any trouble.

I don't like that "sometimes" it works and "sometimes" not. I am team consistency. What do we do now? :D

cc @lm4649

Reply:
Hello @Julien00859 

That is indeed not great, people would probably expect this widget to be consistent with all possible field values, selections, booleans, integers, text etc.

I'm not sure myself what are the limitations of the context based on field values and its interaction with the props updating, that would be something to ask to the JS team probably.

I'm guessing they will have the final approval and r+ on this PR, so they will check this anyway?

As far as I'm concerned, it's LGTM right now, except for that special use case.

Reply:
problem was `edit="0"` in the template :facepalm: